tracee-rules/signatures/golang/stdio_over_socket.go

```diff
@@ -73,20 +73,24 @@ func (sig *stdioOverSocket) OnEvent(e types.Event) error {
 			return err
 		}
 
-		ip := ""
-
 		if connectData.SaFamily == "AF_INET" {
-			ip = connectData.SinAddr
+
+			_, pidExists := sig.processSocketIp[pid]
+			if !pidExists {
+				sig.processSocketIp[pid] = make(map[int]string)
+			}
+
+			sig.processSocketIp[pid][sockfd] = connectData.SinAddr
+
 		} else if connectData.SaFamily == "AF_INET6" {
-			ip = connectData.SinAddr6
-		}
 
-		_, pidExists := sig.processSocketIp[pid]
-		if !pidExists {
-			sig.processSocketIp[pid] = make(map[int]string)
-		}
+			_, pidExists := sig.processSocketIp[pid]
+			if !pidExists {
+				sig.processSocketIp[pid] = make(map[int]string)
+			}
```

Comment:
These lines are the same as lines 78-81 and can be added once before this if statement

Reply:
but then it will create a PID entry in the map for other socket types. I think this is what Roi wanted to fix here, no?

Reply:
No. Previously, non inet sockfds where saved as well, and this is not the case if you create a new map for the pid (which is empty). To avoid creating an empty map, it is possible to put the dup code once in an if (inet or inet6)...